src/OpenRiaServices.EntityFramework/Framework/DbDomainService.cs

```diff
@@ -304,5 +304,68 @@ private static async Task SetChangeSetConflictsAsync(Dictionary<DbEntityEntry, C
                 }
             }
         }
+
+        /// <summary>
+        /// Mark the entity as added
+        /// </summary>
+        /// <typeparam name="TEntity">Entity type</typeparam>
+        /// <param name="entity">The modified entity</param>
+        /// <param name="dbSet">The db set</param>
+        protected void AttachAsAdded<TEntity>(TEntity entity, DbSet<TEntity> dbSet = null)
+            where TEntity : class
+        {
+            EntityEntry<TEntity> entityEntry = DbContext.Entry(entity);
+
+            if (entityEntry.State != EntityState.Detached)
+            {
+                entityEntry.State = EntityState.Added;
```

Comment:
I wounder if there is an actual difference between calling add and setting state in efcore (I should probably have a look).

If it does the same then we could always set state on entry and remove the dbset parameter